src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,194 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Object[] listAsArray = list.toArray();
+        int numNew = listAsArray.length;
+        if (numNew == 0) {
+            return;
+        }
+
+        for (Object object : listAsArray) {
+            T nodeElement = (T) object;
+            Node<T> newNode = new Node<>(last, nodeElement, null);
+            if (last == null) {
+                first = newNode;
+            } else {
+                last.next = newNode;
+            }
+            last = newNode;
+        }
+        size += numNew;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
```

Comment:
This is code repeating. You always check the index before getting node by index. What you should do is move the check logic INSIDE the get method.